fml/status.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_STATUS_H_
+#define FLUTTER_FML_STATUS_H_
+
+#include "flutter/fml/string_view.h"
+
+namespace fml {
+
+/// Google's canonical list of errors, should be in sync with:
+/// https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto
+enum class StatusCode : int {
```

Comment:
We aren't going to be stuffing these inside messages sent over the wire. This enum doesn't need to be typed.

Reply:
The spec for Status specifically requires these are int's because of the contract of things like raw_code()

Reply:
We don't need `raw_code` either do we? We aren't going to be using these in messages over the wire.

Reply:
Well I was thinking it would be helpful if someone wanted to define their own error code.  Maybe we don't want that, I removed it.